apollo-router/tests/integration/query_planner.rs

```diff
@@ -0,0 +1,443 @@
+use std::path::PathBuf;
+
+use crate::integration::common::graph_os_enabled;
+use crate::integration::IntegrationTest;
+
+const PROMETHEUS_METRICS_CONFIG: &str = include_str!("telemetry/fixtures/prometheus.router.yaml");
+const LEGACY_QP: &str = "experimental_query_planner_mode: legacy";
+const NEW_QP: &str = "experimental_query_planner_mode: new";
+const BOTH_QP: &str = "experimental_query_planner_mode: both";
+const BOTH_BEST_EFFORT_QP: &str = "experimental_query_planner_mode: both_best_effort";
+
+#[tokio::test(flavor = "multi_thread")]
+async fn fed1_schema_with_legacy_qp() {
+    let mut router = IntegrationTest::builder()
+        .config(LEGACY_QP)
+        .supergraph("../examples/graphql/local.graphql")
+        .build()
+        .await;
+    router.start().await;
+    router.assert_started().await;
+    router.execute_default_query().await;
+    router.graceful_shutdown().await;
+}
+
+#[tokio::test(flavor = "multi_thread")]
+async fn fed1_schema_with_new_qp() {
+    let mut router = IntegrationTest::builder()
+        .config(NEW_QP)
+        .supergraph("../examples/graphql/local.graphql")
+        .build()
+        .await;
+    router.start().await;
+    router
+        .assert_log_contains(
+            "could not create router: \
+             The supergraph schema failed to produce a valid API schema: \
+             Supergraphs composed with federation version 1 are not supported.",
+        )
+        .await;
+    router.assert_shutdown().await;
+}
+
+#[tokio::test(flavor = "multi_thread")]
+async fn fed1_schema_with_both_qp() {
+    let mut router = IntegrationTest::builder()
+        .config(BOTH_QP)
+        .supergraph("../examples/graphql/local.graphql")
+        .build()
+        .await;
+    router.start().await;
+    router
+        .assert_log_contains(
+            "could not create router: \
+             The supergraph schema failed to produce a valid API schema: \
+             Supergraphs composed with federation version 1 are not supported.",
+        )
+        .await;
+    router.assert_shutdown().await;
+}
+
+#[tokio::test(flavor = "multi_thread")]
+async fn fed1_schema_with_both_best_effort_qp() {
```

Comment:
added reload to both_best_effort integration tests for all 3 features (fed1, context, progressive overrides) so we can also check that the log differs between setting the config to new and both_best_effort